### ISSUE TITLE
Implementation of Basic04 does not correcly output  B04_NO_RESPONSE 

### ISSUE THREAD

**Issue by matsduf** [zonemaster/zonemaster-engine#1104]
lame.dufberg.se is intentionally delegated to two name servers that are not name servers. 
```
;; AUTHORITY SECTION:
lame.dufberg.se.	7200	IN	NS	ns2.lame.dufberg.se.
lame.dufberg.se.	7200	IN	NS	ns1.lame.dufberg.se.

;; ADDITIONAL SECTION:
ns2.lame.dufberg.se.	7200	IN	A	188.126.83.221
ns1.lame.dufberg.se.	7200	IN	A	159.253.30.207
```

[Basic04] should catch that and return B04_NO_RESPONSE (WARNING) unless there is a DNS response, which there is not, but there is no such message:
```
$ zonemaster-cli lame.dufberg.se --test basic/basic04 --level info --raw
   0.00 INFO      UNSPECIFIED    GLOBAL_VERSION   version=v4.5.1
```

Inspecting with level DEBUG the following can be found:
```
   0.05 DEBUG     BASIC04        EXTERNAL_QUERY   flags={"class":"IN"}; ip=188.126.83.221; name=lame.dufberg.se; type=SOA
  10.11 DEBUG     BASIC04        LOOKUP_ERROR   class=IN; domain=lame.dufberg.se; message=Could not send or receive, because of network error.; ns=ns2.lame.dufberg.se/188.126.83.221; type=SOA
  10.12 DEBUG     BASIC04        EXTERNAL_QUERY   flags={"class":"IN"}; ip=159.253.30.207; name=lame.dufberg.se; type=SOA
  20.21 DEBUG     BASIC04        LOOKUP_ERROR   class=IN; domain=lame.dufberg.se; message=Could not send or receive, because of network error.; ns=ns1.lame.dufberg.se/159.253.30.207; type=SOA
(...)
  20.71 DEBUG     BASIC04        EXTERNAL_QUERY   flags={"class":"IN"}; ip=188.126.83.221; name=ns1.lame.dufberg.se; type=A
  20.71 DEBUG     BASIC04        IS_BLACKLISTED   class=IN; dnssec=false; message=Server transport has been blacklisted due to previous failure; name=ns1.lame.dufberg.se; ns=ns2.lame.dufberg.se/188.126.83.221; proto=UDP; type=A
  20.71 DEBUG     BASIC04        EXTERNAL_QUERY   flags={"class":"IN"}; ip=159.253.30.207; name=ns1.lame.dufberg.se; type=A
  20.71 DEBUG     BASIC04        IS_BLACKLISTED   class=IN; dnssec=false; message=Server transport has been blacklisted due to previous failure; name=ns1.lame.dufberg.se; ns=ns1.lame.dufberg.se/159.253.30.207; proto=UDP; type=A
(...)
  20.78 DEBUG     BASIC04        EXTERNAL_QUERY   flags={"class":"IN"}; ip=188.126.83.221; name=ns1.lame.dufberg.se; type=AAAA
  20.78 DEBUG     BASIC04        IS_BLACKLISTED   class=IN; dnssec=false; message=Server transport has been blacklisted due to previous failure; name=ns1.lame.dufberg.se; ns=ns2.lame.dufberg.se/188.126.83.221; proto=UDP; type=AAAA
  20.78 DEBUG     BASIC04        EXTERNAL_QUERY   flags={"class":"IN"}; ip=159.253.30.207; name=ns1.lame.dufberg.se; type=AAAA
  20.78 DEBUG     BASIC04        IS_BLACKLISTED   class=IN; dnssec=false; message=Server transport has been blacklisted due to previous failure; name=ns1.lame.dufberg.se; ns=ns1.lame.dufberg.se/159.253.30.207; proto=UDP; type=AAAA
(...)
  20.84 DEBUG     BASIC04        EXTERNAL_QUERY   flags={"class":"IN"}; ip=188.126.83.221; name=ns2.lame.dufberg.se; type=A
  20.84 DEBUG     BASIC04        IS_BLACKLISTED   class=IN; dnssec=false; message=Server transport has been blacklisted due to previous failure; name=ns2.lame.dufberg.se; ns=ns2.lame.dufberg.se/188.126.83.221; proto=UDP; type=A
  20.84 DEBUG     BASIC04        EXTERNAL_QUERY   flags={"class":"IN"}; ip=159.253.30.207; name=ns2.lame.dufberg.se; type=A
  20.84 DEBUG     BASIC04        IS_BLACKLISTED   class=IN; dnssec=false; message=Server transport has been blacklisted due to previous failure; name=ns2.lame.dufberg.se; ns=ns1.lame.dufberg.se/159.253.30.207; proto=UDP; type=A
(...)
  20.90 DEBUG     BASIC04        EXTERNAL_QUERY   flags={"class":"IN"}; ip=188.126.83.221; name=ns2.lame.dufberg.se; type=AAAA
  20.90 DEBUG     BASIC04        IS_BLACKLISTED   class=IN; dnssec=false; message=Server transport has been blacklisted due to previous failure; name=ns2.lame.dufberg.se; ns=ns2.lame.dufberg.se/188.126.83.221; proto=UDP; type=AAAA
  20.90 DEBUG     BASIC04        EXTERNAL_QUERY   flags={"class":"IN"}; ip=159.253.30.207; name=ns2.lame.dufberg.se; type=AAAA
  20.90 DEBUG     BASIC04        IS_BLACKLISTED   class=IN; dnssec=false; message=Server transport has been blacklisted due to previous failure; name=ns2.lame.dufberg.se; ns=ns1.lame.dufberg.se/159.253.30.207; proto=UDP; type=AAAA
```
 
Trying to query the name server (that are not name servers) result in no DNS response:
```
$ dig @188.126.83.221 lame.dufberg.se
;; communications error to 188.126.83.221#53: connection refused

$ dig @159.253.30.207 lame.dufberg.se

; <<>> DiG 9.18.2 <<>> @159.253.30.207 lame.dufberg.se
; (1 server found)
;; global options: +cmd
;; connection timed out; no servers could be reached
```

[Basic04]: https://github.com/zonemaster/zonemaster/blob/master/docs/specifications/tests/Basic-TP/basic04.md

**Comment by tgreenx** [comment]
I can see the problem: in that case, [Method4&5](https://github.com/zonemaster/zonemaster-engine/blob/master/lib/Zonemaster/Engine/Test/Basic.pm#L530) returns an empty array, which means that both glue and nameservers for the zone are missing, preventing the test case from pursuing and outputting any messages.

I see two approaches here (second is better, imho) :
 - I can have a quick fix using `B04_NO_RESPONSE` as message tag but in those cases $ns will never be filled, meaning it will show like this:
  ```
  zonemaster-cli lame.dufberg.se --show-testcase --test basic/basic04 --level INFO
  Seconds Level     Testcase       Message
  ======= ========= ============== =======
     0.00 INFO      UNSPECIFIED    Using version v4.5.1 of the Zonemaster engine.
    21.03 WARNING   BASIC04        Nameserver {ns} does not respond over neither UDP nor TCP.
  ```
- We have the specification updated with a new message tag similar to the one found in Basic02  `NO_GLUE_PREVENTS_NAMESERVER_TESTS` (it checks for glue only, and is run after Basic04), adapted for this test case.

**Comment by matsduf** [comment]
> I see two approaches here (second is better, imho) :
> 
>     * I can have a quick fix using `B04_NO_RESPONSE` as message tag but in those cases $ns will never be filled, meaning it will show like this:

That should be OK until we have better methods. That is a reasonable work-around.

>     * We have the specification updated with a new message tag similar to the one found in Basic02  `NO_GLUE_PREVENTS_NAMESERVER_TESTS` (it checks for glue only, and is run after Basic04), adapted for this test case.

I see not point of duplicating the message. The message from Basic02 does not list the name or IP address of the name server. And the message is confusing anyway:

Raw:
```
  20.75 CRITICAL  BASIC02        NO_GLUE_PREVENTS_NAMESERVER_TESTS   
```
With human text:
```
  20.76 CRITICAL  BASIC02        No NS records for tested zone from parent. NS tests skipped.
```

There are PRs with updated specifications of Basic01 and Basic02 that should give better messages.

**Comment by tgreenx** [comment]
Solved by #1111
